File: app/src/main/java/com/kaciula/archiman/util/CrashlyticsTree.java
package com.kaciula.archiman.util;

import android.util.Log;

import com.crashlytics.android.Crashlytics;

import timber.log.Timber;

public class CrashlyticsTree extends Timber.Tree {

    @Override
    protected void log(int priority, String tag, String message, Throwable t) {
        if (priority == Log.VERBOSE || priority == Log.DEBUG) {
            return;
        }

        Crashlytics.log(message);

        if (t != null) {
            if (priority == Log.ERROR) {
                Crashlytics.logException(t);
            }
        }
    }
}

File: app/src/main/java/com/kaciula/archiman/infrastructure/CrashlyticsTree.kt
package com.kaciula.archiman.infrastructure

import android.util.Log
import com.crashlytics.android.Crashlytics
import timber.log.Timber

class CrashlyticsTree : Timber.Tree() {

    override fun log(priority: Int, tag: String, message: String, t: Throwable?) {
        if (priority == Log.VERBOSE || priority == Log.DEBUG) {
            return
        }

        Crashlytics.log(message)

        if (t != null && priority == Log.ERROR) {
            Crashlytics.logException(t)
        }
    }
}